### DISCUSSION TITLE
Looking for media table on landlord instead of tenant database

### DISCUSSION THREAD

**Discussion by cverster** [spatie/laravel-medialibrary#2470] [Q&A]
Hi all

I've implemented both Spatie Medialibrary and Spatie multitenancy on my application, and I've run into an issue.
I use media library to set avatars and thumbnails for my users, and capture their images in my registration requests. However, I'm currently getting the following error:

`Base table or view not found: 1146 Table 'landlord.media' doesn't exist (SQL: select max(`order_column`) as aggregate from `media`)`

For some reason, the program is looking for the media table on my landlord database (which is my default) instead of my tenant database (which my model uses). I've set the UsesTenantConnection trait on the model in question, so I would expect it to look at the tenant database for the media table?

Additionally, I'm only experiencing this issue on my prod server, not my dev server, and my prod server has many more media entries. It could not have something to do with the amount of media stored in the table right? This was a non-issue, and it seems to have "broken by itself", as many people were able to successfully upload previously.

How can I get the media library to always honour the tenant connection instead of the landlord connection?

**Comment by Mudsr** [comment]
Just two days ago i was facing this issue i am using hyn tenancy package. probably in some vendor classes they forgot to use the media model's namespace value from config/media-library.php or may be they had reason to hardcode the media model's namespace. Here is the solution that implemented. I tracked the class where they were calling the default medial model then created and alias for that and in the new class i replaced the namespace with the media model that i extended to use tenant connection. For me the culprit class was spatie\MediaLibraryPro\Dto\ViewMediaItem.php.
![image](https://user-images.githubusercontent.com/62287001/121776878-9f26da80-cba8-11eb-9def-b77dc6ff1b90.png)

**Reply by cverster** [reply]
Hi Mudsr

Thanks for this! In my case I'm just extending HasMediaTrait on my model in question and calling normal spatie functions on that, but I'm interested to know how you extended the media class to use the tenant connection?

**Reply by Mudsr** [reply]
Ah i see. I assumed you already tried all those options. I am not sure about the tenancy package that you are using but in hyn tenancy package we get a trait that we have to include in model to make it use the tenant connection. Here is the screenshot of how i extended the media model. After extending the media model you can just tell the medial-library to use this model in config/media-library.php 
![image](https://user-images.githubusercontent.com/62287001/122040757-a42a9a80-cdf1-11eb-9277-4f4cb69f146c.png)
